site/app/controllers/AutogradingStatusController.php

```diff
@@ -44,6 +45,36 @@ public function getProgress(): JsonResponse {
         return JsonResponse::getSuccessResponse($info);
     }
 
+    /**
+     * Attempts to read the stack trace and find any error message
+     * @Route("/autograding_status/get_stack", methods={"GET"})
+     * @return JsonResponse
+     */
+    public function getErrorStackTrace(): JsonResponse {
+        if (empty($this->core->getQueries()->getInstructorLevelAccessCourse($this->core->getUser()->getId()))) {
+            return JsonResponse::getFailResponse("You do not have access to this endpoint.");
+        }
+        $stack_trace_path = FileUtils::joinPaths($this->core->getConfig()->getSubmittyPath(), "logs", "autograding_stack_traces");
+        if (!is_readable($stack_trace_path)) {
+            return JsonResponse::getFailResponse("Could not access the stack trace path.");
+        }
+        $files = scandir($stack_trace_path, SCANDIR_SORT_DESCENDING);
```

Comment:
If we only show the last 7 stack traces, why do we want to return all of them here? What happens if an instructor has hundreds of stack traces that in total weigh several GB? We should probably do the limiting of number of returned files _here_ instead of within the JS code and avoid sending unnecessary bytes over the wire, slowing down load of the page.